tracing-appender/src/non_blocking.rs

```diff
@@ -277,7 +277,16 @@ impl Drop for WorkerGuard {
                 // when the `Worker` calls `recv()` on a zero-capacity channel. Use `send_timeout`
                 // so that drop is not blocked indefinitely.
                 // TODO: Make timeout configurable.
-                let _ = self.shutdown.send_timeout((), Duration::from_millis(1000));
+                match self.shutdown.send_timeout((), Duration::from_millis(1000)) {
+                    Err(SendTimeoutError::Disconnected(_)) => (),
```

Comment:
if we get the `Disconnected` error, this means that the `Receiver` side of the channel has already been dropped...but that doesn't necessarily mean that the worker thread has terminated yet. should we still call `join` on the `JoinHandle` in this case?

Reply:
> if we get the Disconnected error, this means that the Receiver side of the channel has already been dropped...but that doesn't necessarily mean that the worker thread has terminated yet.

right, this can be racy also

If `Disconnected` can occur only during worker thread (which owns `Receiver`) destruction, I think it is safe to use `join` here.
Went ahead and merged `Disconnected` with `Ok` case.